drivers/i2c/i2c_bitbang.c

```diff
@@ -76,9 +82,30 @@ int i2c_bitbang_get_config(struct i2c_bitbang *context, uint32_t *config)
 	return 0;
 }
 
+#ifdef CONFIG_I2C_GPIO_CLOCK_STRETCHING
+static void i2c_wait_clock_stretching(struct i2c_bitbang *context)
+{
+	uint32_t timeout = context->delays[T_CLK_STR];
+	uint32_t start = k_cycle_get_32();
+
+	do {
```

Comment:
We have a common macro for this sort of thing in zephyr/sys/util.h

```c
bool scl_ready = WAIT_FOR(context->io->get_scl(context->io_context) != 0, timeout_us, ;);
```

if the macro does not work for you, let me know